sql/rename.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: XisiHuang (cockhuangxh@163.com)
+
+package sql
+
+import (
+	"fmt"
+
+	"github.com/cockroachdb/cockroach/client"
+	"github.com/cockroachdb/cockroach/proto"
+	"github.com/cockroachdb/cockroach/sql/parser"
+	"github.com/cockroachdb/cockroach/structured"
+)
+
+// RenameDatabase alters a databsase name
+// Privileges: "root" user.
+//   Notes: postgres requires superuser, db owner, or "CREATEDB".
+//          mysql >= 5.1.23 does not allow database renames
+func (p *planner) RenameDatabase(n *parser.RenameDatabase) (planNode, error) {
+	if n.Name == "" || n.NewName == "" {
+		return nil, errEmptyDatabaseName
+	}
+
+	if n.Name == n.NewName {
+		//noop
+		return &valuesNode{}, nil
+	}
+
+	dbDesc, err := p.getDatabaseDesc(string(n.Name))
+	if err != nil {
+		return nil, err
+	}
+
+	if !dbDesc.HasPrivilege(p.user, parser.PrivilegeWrite) {
```

Comment:
as mentioned before, this should check for "root" user, just like CreateDatabase.